util/WorkQueue.hpp

```diff
@@ -76,12 +77,28 @@ class WorkQueue
             ::SetThreadpoolCallbackPool(&m_threadpoolEnv, m_threadPool.get());
         }
 
+        ~ThreadPool()
+        {
+            Reset();
+        }
+
+        ThreadPool(const ThreadPool&) = delete;
+        ThreadPool(ThreadPool&&) = delete;
+        ThreadPool& operator=(const ThreadPool&) = delete;
+        ThreadPool& operator=(ThreadPool&&) = delete;
+
         wil::unique_threadpool_work CreateWork(PTP_WORK_CALLBACK callback, void* context)
         {
             wil::unique_threadpool_work work(::CreateThreadpoolWork(callback, context, &m_threadpoolEnv));
             THROW_LAST_ERROR_IF_NULL(work.get());
             return work;
         }
+
+        void Reset() noexcept
+        {
+            m_threadPool.reset();
+            ::DestroyThreadpoolEnvironment(&m_threadpoolEnv);
```

Comment:
can we just declare a

using unique_threadpool_environment = wil::unique_struct<TP_CALLBACK_ENVIRON, decltype(&::DestroyThreadpoolEnvironment), ::DestroyThreadpoolEnvironment>;

then reset() that?